src/services/gptService.ts

```diff
@@ -1,18 +1,75 @@
 import { v4 as uuid } from "uuid";
-import { Moment } from "../schema/app_schema.js";
 import { AzureOpenAI } from "openai";
 import axios from "axios";
 import { ChatCompletionCreateParamsNonStreaming } from "openai/resources/index.mjs";
 import { AccountInfo } from "@azure/msal-browser";
+import { Moment } from "../schema/app_schema.js";
+
+export class GPTService {
+	private static _instance: GPTService;
+	private static get instance(): GPTService {
+		if (GPTService._instance === undefined) {
+			throw new Error("GPTService not initialized");
+		}
+		return GPTService._instance;
+	}
+
+	public static initialize(account: AccountInfo) {
+		if (GPTService._instance !== undefined) {
+			throw new Error("Only allowed to initialize GPTService once");
+		}
+		this._instance = new GPTService(account);
+	}
+
+	private gptPrompter: ReturnType<typeof createSessionPrompter> | undefined;
+	private constructor(private readonly account: AccountInfo) {
+		this.createPrompterIfNeeded();
+	}
+
+	private createPrompterIfNeeded() {
+		if (this.gptPrompter === undefined) {
+			try {
+				this.gptPrompter = createSessionPrompter(this.account);
+			} catch (e) {
+				console.error("Failed to create AI prompter. Please try again.", e);
+			}
+		}
+	}
+
+	public static async prompt(momentDescription: string): Promise<Moment[]> {
+		const instance = GPTService.instance;
+		instance.createPrompterIfNeeded();
```

Comment:
Is this redundant with the call in the constructor?

Reply:
The call may fail for whatever reason, so we re-try it. A possible improvement could be to have some background timer running the refresh instead.

Reply:
Looks to me like it's synchronous and deterministic, so probably is overkill.  Can clean it up later though.